Updated the control variable from the transformations loop

elvanov opened the pull request:
Seq_len works with both, tibble and data frame objects.

elvanov committed:
Updated the control variable from the transformations loop 

Seq_len works with both, tibble and data frame objects.

tnaake commented:
Hi @elvanov ,
many thanks for your PR. Looks good - could you also still add some unit tests for the `tibble` case?

tnaake commented on R/structural.R:
```diff
@@ -211,7 +211,7 @@ structural <- function(x, transformation, var = character(),
     ## iterate through each column and check if the "mass" is in the interval
     ## defined by the m/z value and ppm
-    for (i in seq_along(transformation[, "mass"])) {
+    for (i in seq_len(nrow(transformation))) {
```
Could you check if this works properly on `data.frame`/`tibble` (i.e. write some unit tests for these cases)?

elvanov replied:
Hello @tnaake, I observed that the unit tests you already had would fail if the transformation table was a tibble object, so I added a conversion to data.frame at the beginning of the structural function. And I also added the corresponding unit tests.
Thanks!

elvanov committed:
Improved structural function to work with transformation tables that are tibble objects, and added the corresponding unit tests.

tnaake commented:
Almost there, there is still a NAMESPACE issue. Could you solve this? 
I think an `#' @importFrom tibble is_tibble` at an appropriate spot in `structural` will fix this.

elvanov committed:
Fixed namespace bug.

elvanov commented:
I guess the issue should be solved now.

tnaake commented:
Hi @elvanov ,

thanks for the update. Unfortunately, the tests do not pass. Could you have another look?

elvanov committed:
Fixed bug that made some tests fail.

elvanov commented:
Hello @tnaake,

I put the unit tests that use the tibble object before those that use the data frame. All the tests should pass now.

elvanov committed:
Added the transformation table as tibble object use case to the rtCorrection function.

elvanov commented:
Hi @tnaake,

I noticed that the transformation table is also used in the rtCorrection function, so I also added the tibble use case there, and I added the corresponding unit tests.

tnaake reviewed (commented):
Great, I will merge the PR!

Many thanks!
